src/dlfcn.c

```diff
@@ -467,7 +472,170 @@ char *dlerror( void )
     return error_buffer;
 }
 
-#ifdef SHARED
+/* taken from http://bandido.ch/programming/Import_Address_Table_Hooking.pdf */
+static IMAGE_IMPORT_DESCRIPTOR* get_import_address_table( HMODULE module, DWORD *size )
+{
+    IMAGE_DOS_HEADER* dosHeader = (IMAGE_DOS_HEADER*)module;
+    if ( dosHeader->e_magic != 0x5A4D )
+        return NULL;
+    IMAGE_OPTIONAL_HEADER* optionalHeader = (IMAGE_OPTIONAL_HEADER*)

```

Comment:
Compilation under WINDDK is failing on following error:
```
dlfcn.c(481) : error C2275: 'IMAGE_OPTIONAL_HEADER' : illegal use of this type as an expression
1> c:\winddk\7600.16385.win7_wdk.100208-1538\inc\api\winnt.h(11084) : see declaration of 'IMAGE_OPTIONAL_HEADER'
```

Reason is that variables needs to be delared at beginning.

Following change is fixing it (I tested it):
```c
{
    IMAGE_DOS_HEADER *dosHeader;
    IMAGE_OPTIONAL_HEADER *optionalHeader;

    dosHeader = (IMAGE_DOS_HEADER *) module;
    if( dosHeader->e_magic != 0x5A4D )
        return NULL;
    optionalHeader = (IMAGE_OPTIONAL_HEADER *) ((BYTE *) module + dosHeader->e_lfanew + 24);
...
```